aiida/orm/implementation/django/comment.py

```diff
@@ -7,126 +7,135 @@
-                    "database")
-            elif ccount == 0:
-                raise NotExistent("No comments were found with the given "
-                                 "arguments")
-
-            self.dbcomment = res[0]
+from aiida.backends.djsite.db import models
+from aiida.common import exceptions
+from aiida.common import utils
 
-    @property
-    def pk(self):
-        return self.dbnode.pk
+from ..comments import BackendComment, BackendCommentCollection
+from .utils import ModelWrapper
+from . import entities
+from . import users
 
-    @property
-    def id(self):
-        return self.dbnode.pk
 
-    @property
-    def to_be_stored(self):
-        return self.dbcomment.pk is None
+class DjangoComment(entities.DjangoModelEntity[models.DbComment], BackendComment):
+    """Comment implementation for Django."""
 
-    @property
-    def uuid(self):
-        return six.text_type(self.dbcomment.uuid)
+    MODEL_CLASS = models.DbComment
 
-    def get_ctime(self):
-        return self.dbcomment.ctime
+    def __init__(self, backend, node, user, content=None):
+        """
+        Construct a DjangoComment.
 
-    def set_ctime(self, val):
-        self.dbcomment.ctime = val
-        if not self.to_be_stored:
-            self.dbcomment.save()
+        :param node: a Node instance
+        :param user: a User instance
+        :param content: the comment content
+        :return: a Comment object associated to the given node and user
+        """
+        super(DjangoComment, self).__init__(backend)
+        utils.type_check(user, users.DjangoUser)  # pylint: disable=no-member
+        self._dbmodel = ModelWrapper(models.DbComment(dbnode=node.dbnode, user=user.dbmodel, content=content))
 
-    def get_mtime(self):
-        return self.dbcomment.mtime
+    def store(self):
+        """Can only store if both the node and user are stored as well."""
+        if self._dbmodel.dbnode.id is None or self._dbmodel.user.id is None:
+            raise exceptions.ModificationNotAllowed
 
-    def set_mtime(self, val):
-        self.dbcomment.mtime = val
-        if not self.to_be_stored:
-            self.dbcomment.save()
+        super(DjangoComment, self).store()
 
-    def get_user(self):
-        return self.dbcomment.user
+    @property
+    def uuid(self):
+        return six.text_type(self._dbmodel.uuid)
 
-    def set_user(self, val):
-        self.dbcomment.user = val
-        if not self.to_be_stored:
-            self.dbcomment.save()
+    @property
+    def ctime(self):
+        return self._dbmodel.ctime
 
-    def get_content(self):
-        return self.dbcomment.content
+    @property
+    def mtime(self):
+        return self._dbmodel.mtime
 
-    def set_content(self, val):
-        self.dbcomment.content = val
-        if not self.to_be_stored:
-            self.dbcomment.save()
+    def set_mtime(self, value):
+        self._dbmodel.mtime = value
+        if self.is_stored:
```

Comment:
This is not needed.  The `ModelWrapper` does this for all attributes anyway.  Sorry I missed it the first time.